src/app/filter-by-regex.pipe.spec.ts

```diff
@@ -1,8 +1,46 @@
 import { FilterByRegexPipe } from './filter-by-regex.pipe';
 
 describe('FilterByRegexPipe', () => {
+
+  const mockTemplate = {
+    href: 'https://www.mocklink.com/page',
+    host: 'www.mocklink.com',
+    domId: 0,
+    tagName: 'A',
+  };
+
+  const mockList = [];
```

Comment:
Initializing an input array this way can make it hard for reader to read it. So it's better to avoid it unless necessary. It's not obvious how many has "visible" true.

I would do it by hand instead, like

const mockList = [{
   href: 'https://www.mocklink.com/page/1'
   host: 'www.mocklink.com',
   domId: 0,
   visible: true
   tagName: 'A'
}]